### ISSUE TITLE
Using sigaction won't compile on Windows 

### ISSUE THREAD

**Issue by taanders** [libimobiledevice/idevicerestore#283]
Using sigaction won't compile on Windows:
https://github.com/libimobiledevice/idevicerestore/commit/3050e61588bfc79b3b360fa25db32c81e3264efb

I think it needs to be done like this way on Windows side:
https://stackoverflow.com/questions/32389905/sigaction-and-porting-linux-code-to-windows

```
    signal(SIGINT, handle_signal);
    signal(SIGTERM, handle_signal);
    signal(SIGABRT, handle_signal)
```

**Comment by akhileshkrsingh1** [comment]
You're correct. sigaction is typically used in Unix-like systems (such as Linux, macOS, etc.) for handling signals. Windows doesn't use signals in the same way Unix does, so sigaction isn't available on Windows.

On Windows, you'd typically use different mechanisms for similar purposes, such as Windows signals or structured exception handling (SEH). For example, you might use SetUnhandledExceptionFilter to catch unhandled exceptions.
